scripts/rotating_camera/controllers/rotating_camera/rotating_camera.c

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 1996-2020 Cyberbotics Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <stdio.h>
+#include <math.h>
+#include <webots/robot.h>
+#include <webots/supervisor.h>
+
+#define TOL 1e-6
+#define ANGLE_REV 6.28
+#define CW  0
+#define CCW 1
+#define UP 0
+#define DOWN 1
+
+// Global variables
+WbNodeRef root, nodes;
+WbNodeRef nodeViewPoint;
+WbNodeRef nodeRobot;
+
+WbFieldRef children;
+WbFieldRef fieldViewPointPos;
+WbFieldRef fieldViewPointOrient;
+WbFieldRef fieldRobotTranslation;
+
+// The viewpoint turn around the robot in a circle
+//  o robotPos : Specifies the robot position.
+//  o direction : CW (Turn ClockWise, default) or CCW (turn CounterClockWise).
+//  o nTurn : Make the number of complet revolution defined.
+//  o radius : Set the distance between the robot and the camera.
+//  o heightStep : Height achieved during a complet revolution.
+//  o angleEnd : Make an arc of a circle. Set to 6.28 for complet revolution.
+
+void turningView(double *robotPos, int rotationDirection, int verticalDirection, int nTurn, float radius, float heightStep, double angleEnd) {
```

Comment:
case/underscore is wrong.
This function should be declared as `static`.
I would call it revolving instead of turning... This is more precise.